interfaces/tests/timer_int/timer_int.c

```diff
@@ -0,0 +1,301 @@
+{
+    ///arrange
+    TIMER_HANDLE timer = timer_create_new();
+    ASSERT_IS_NOT_NULL(timer);
+
+    // sleep 1s
+    ThreadAPI_Sleep(1000);
+
+    ///act
+    double result = timer_get_elapsed(timer);
+
+    ///assert
+    ASSERT_IS_TRUE(result > 0.5);
+    ASSERT_IS_TRUE(result < 1.5);
+
+    /// cleanup
+    timer_destroy(timer);
+}
+
+/* Tests_SRS_TIMER_01_007: [ Otherwise timer_get_elapsed shall return the time difference in seconds between the current time and the start time of the timer. ]*/
+TEST_FUNCTION(timer_get_elapsed_with_start_succeeds)
+{
+    ///arrange
+    TIMER_HANDLE timer = timer_create_new();
+    ASSERT_IS_NOT_NULL(timer);
+
+    // sleep 3s
+    ThreadAPI_Sleep(3000);
+    // start again
+    ASSERT_ARE_EQUAL(int, 0, timer_start(timer));
+
+    ThreadAPI_Sleep(1000);
+
+    ///act
+    double result = timer_get_elapsed(timer);
+
+    ///assert
+    /// giving it a wide tolerance
+    ASSERT_IS_TRUE(result > 0.5);
+    ASSERT_IS_TRUE(result < 1.5);
+
+    /// cleanup
+    timer_destroy(timer);
+}
+
+/* timer_get_elapsed_ms */
+
+/* Tests_SRS_TIMER_01_008: [ if timer is NULL, timer_get_elapsed_ms shall return a negative value. ]*/
+TEST_FUNCTION(timer_get_elapsed_ms_with_NULL_timer_fails)
+{
+    ///arrange
+
+    ///act
+    double result = timer_get_elapsed(NULL);
+
+    ///assert
+    ASSERT_ARE_EQUAL(double, -1.0, result);
+}
+
+/* Tests_SRS_TIMER_01_009: [ Otherwise timer_get_elapsed_ms shall return the time difference in milliseconds between the current time and the start time of the timer. ]*/
+TEST_FUNCTION(timer_get_elapsed_ms_with_implicit_start_succeeds)
+{
+    ///arrange
+    TIMER_HANDLE timer = timer_create_new();
+    ASSERT_IS_NOT_NULL(timer);
+
+    // sleep 1s
+    ThreadAPI_Sleep(2000);
+
+    ///act
+    double result = timer_get_elapsed_ms(timer);
+
+    ///assert
+    /// giving it a wide tolerance
+    ASSERT_IS_TRUE(result > 1500);
+    ASSERT_IS_TRUE(result < 2500);
+
+    /// cleanup
+    timer_destroy(timer);
+}
+
+/* Tests_SRS_TIMER_01_009: [ Otherwise timer_get_elapsed_ms shall return the time difference in milliseconds between the current time and the start time of the timer. ]*/
+TEST_FUNCTION(timer_get_elapsed_ms_with_start_succeeds)
+{
+    ///arrange
+    TIMER_HANDLE timer = timer_create_new();
+    ASSERT_IS_NOT_NULL(timer);
+
+    // sleep 3s
+    ThreadAPI_Sleep(3000);
+    // start again
+    ASSERT_ARE_EQUAL(int, 0, timer_start(timer));
+
+    ThreadAPI_Sleep(1000);
+
+    ///act
+    double result = timer_get_elapsed_ms(timer);
+
+    ///assert
+    /// giving it a wide tolerance
+    ASSERT_IS_TRUE(result > 500);
+    ASSERT_IS_TRUE(result < 1500);
+
+    /// cleanup
+    timer_destroy(timer);
+}
+
+/* timer_global_get_elapsed_ms */
+
+/* Tests_SRS_TIMER_01_010: [ timer_global_get_elapsed_ms shall return the elapsed time in milliseconds from a start time in the past. ]*/
+TEST_FUNCTION(timer_global_get_elapsed_ms_measures_a_seconds)
+{
+    ///arrange
+
+    // sleep 3s
```

Comment:
>/ sleep 3s [](start = 5, length = 10)

not sleeping 3s here :)

Reply:
Fixed.

---
In reply to: [512807043](https://github.com/Azure/azure-c-pal/pull/48#discussion_r512807043) [](ancestors = 512807043)